src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -574,35 +691,104 @@ private void AddSchema(CsdlSchema schema, bool addAnnotations)
             {
                 RegisterElement(container);
             }
+        }
 
-            if (!string.IsNullOrEmpty(schema.Alias))
+        /// <summary>
+        /// Adds out-of-line annotations from the specified schema
+        /// if they are referenced in the <paramref name="includeAnnotationsIndex"/> cache.
+        /// </summary>
+        /// <param name="schema">The schema to add annotations from.</param>
+        /// <param name="schemaWrapper">The <see cref="CsdlSemanticsSchema"/> wrapper for the provided schema</param>
+        /// <param name="includeAnnotationsIndex">Cache of the schema's out-of-line <see cref="IEdmIncludeAnnotations"/>s indexed by the term namespace.
+        /// If this dictionary is non-empty, then only annotations that match
+        /// the references will be added. If it's empty, no annotations will be added. If it's null, all annotations
+        /// will be added unconditionally.</param>
+        private void AddOutOfLineAnnotationsFromSchema(

```

Comment:
@habbes RE: _If it's null, all annotations will be added unconditionally_
Does this happen from the calling method or this (`AddOutOfLineAnnotationsFromSchema`) method?

Reply:
That actually takes place in the `FilterIncludedAnnotations` method, which is called by this method (`AddOutOfLineAnnotationsFromSchema`). The `FilterIncludedAnnotations` method has a similar note in its doc comments.